spring-kafka/src/main/java/org/springframework/kafka/support/serializer/JsonSerializer.java

```diff
@@ -138,6 +141,9 @@ else if (config instanceof String) {
 
 	@Override
 	public byte[] serialize(String topic, T data) {
+		if (data == null) {
+			return null;
+		}
 		try {
 			byte[] result = null;
 			if (data != null) {
```

Comment:
this check for not null is now redundant

Reply:
😄 

Good catch! 
Feel free to provide a contribution on the matter!